File: 2022/day1/js/day1.js
const fs = require("fs");

fs.readFile("../input.txt", "utf8", (err, data) => {
  if (err) {
    console.error(err);
    return;
  }

  // Part 1
  let a = data.split("\n\n").map((item) =>
    item
      .replace(/^\s+|\s+$/g, "")
      .split("\n")
      .reduce((acc, cur) => {
        return acc + Number.parseInt(cur, 10);
      }, 0)
  );
  console.log(Math.max(...a));

  // Part 2
  let b = a;
  let s = 0;
  for (i = 0; i < 3; i++) {
    let max = Math.max(...b);
    s += max;
    b.splice(b.indexOf(max), 1);
  }
  console.log(s);

  return 0;
});

File: 2022/day1/py/day1.py
def main() -> int:
    with open('../input.txt', 'r') as file:

        # Part 1
        a = [x.strip().split('\n') for x in file.read().split('\n\n')]
        b = [eval("+".join(x)) for x in a]
        print(max(b))
        
        # Part 2
        c = b
        s = 0
        for i in range(0,3):
            s += max(c)
            c.remove(max(c))
        print(s)

    return 0

if __name__ == '__main__':
    exit(main())
